pkg/trace/config/config.go

```diff
@@ -268,3 +268,12 @@ func prepareConfig(path string) (*AgentConfig, error) {
 func HasFeature(f string) bool {
 	return strings.Contains(os.Getenv("DD_APM_FEATURES"), f)
 }
+
+// Features returns a list of all the features configured by means of DD_APM_FEATURES.
+func Features() []string {
```

Comment:
I think we need a bit more than this.  We don't want all the tracers to have to bake in knowledge about the defaults of the trace agent.

So we need a map of all the feature flags even those are using the default values.

In the Java tracer, we accomplish this by building all the configuration up into an object rather than accessing the environment directly.  But I'm guessing there isn't something similar in the trace agent?

Reply:
Yeah I think it'd be good to have for the codebase too. Otherwise we don't have a list of features. You mean something like:
```json
features: {
    feature1: true,
    feature2: false,
    // list goes on with all features
}
```
?

Reply:
Yes, basically, we need to know the state of the flags even the ones that a user didn't override.
Otherwise, we have to know which flags exist and their defaults for every tracer version.  We'd like to avoid that sort of coupling if at all possible.  Especially, since we'll have to replicate it in every tracer.

One slight tweak, I wouldn't limit to booleans.  For instance, it would nice if the tracer knew the payload size limit as well.

Reply:
Sure, we can have that in a separate section where we share the agent config.

Reply:
Actually, I don't think it's a good idea to include disabled feature _flags_. In the future, we may remove feature flags and then it might confuse tracers looking for them specifically. The list of features will only include set feature flags.

What you are referring to however, about defaults and config, that we can have in a separate section. Does it make sense?

Reply:
@dougqh checkout the PR description FYI